### PR TITLE
Increase test parallelism to 4

### PR DESCRIPTION
**Pull request by kevingosse**

### CODE REVIEW

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/CustomTestFramework.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 using Xunit.Abstractions;
 using Xunit.Sdk;
 
-[assembly: CollectionBehavior(MaxParallelThreads = 12)]
+[assembly: CollectionBehavior(MaxParallelThreads = 4)]
```
Comment (andrewlock) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/CustomTestFramework.cs:11]: Do we know how many it's using at the moment? i.e. how many virtual CPUs do we have?

Comment (kevingosse) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/CustomTestFramework.cs:11]: afaik, 2

Comment (kevingosse) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/CustomTestFramework.cs:11]: They run on Standard_DS2_v2: https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series

Comment (andrewlock) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/CustomTestFramework.cs:11]: Then I know how to speed up these tests 😉  (`Standard_DS3_v2`)

Comment (kevingosse) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/CustomTestFramework.cs:11]: I wish :D But we have no control over it, short of running our own instances: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#hardware

**Comment by kevingosse** [comment]
This gives no obvious speed up over the default value

**Pull request closed without merging**